src/transforms/deployment.ts

```diff
@@ -1,7 +1,79 @@
 import _ from "lodash";
-import { Context } from "probot/lib/context";
 import issueKeyParser from "jira-issue-key-parser";
-import { JiraDeploymentData } from "../interfaces/jira";
+import {JiraDeploymentData} from "../interfaces/jira";
+import {GitHubAPI} from "probot";
+import {WebhookPayloadDeploymentStatus} from "@octokit/webhooks";
+import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import {Octokit} from "@octokit/rest";
+import {booleanFlag, BooleanFlags} from "../config/feature-flags";
+
+async function getLastSuccessfulDeployCommitSha(
+	owner: string,
+	repoName: string,
+	github: GitHubAPI,
+	deployments: Octokit.ReposListDeploymentsResponseItem[]
+): Promise<string> {
+
+	for (const deployment of deployments) {
+		// Get each deployment status for this environment so we can have their statuses' ids
+		const listDeploymentStatusResponse: Octokit.Response<Octokit.ReposListDeploymentStatusesResponse> = await github.repos.listDeploymentStatuses({
+			owner: owner,
+			repo: repoName,
+			deployment_id: deployment.id
+		});
+
+		// Filter only the successful ones
+		const lastSuccessful: Octokit.ReposListDeploymentStatusesResponseItem[] = listDeploymentStatusResponse.data.filter(deployment => deployment.state === "success");
+		if (lastSuccessful) {
```

Comment:
this will always return true unless the `listDeploymentStatusResponse.data` is undefined which will then break your code.  Are you sure you didn't mean to do `find` instead of `filter`?

Reply:
Does `if(lastSuccessful)` resolve to `true` if the array is empty?

Yeah, I guess using `find` instead of `filter` is more straightforward.

Reply:
Changed to use `find` instead :)